scripts/zones/Attohwa_Chasm/mobs/Xolotls_Sacrifice.lua

```diff
@@ -0,0 +1,21 @@
+-----------------------------------
+-- Area: Attohwa Chasm
+--  Mob: Xolotl's Sacrifice
+-- Note: Pet for Xolotl
+-----------------------------------
+require("scripts/globals/world")
+-----------------------------------
+local entity = {}
+
+entity.onMobRoam = function(mob)
+    local totd = VanadielTOTD()
+
+    if totd ~= xi.time.NIGHT and totd ~= xi.time.MIDNIGHT then
```

Comment:
check if it or its master are actively in a battle too. Have held him in daytime. Claim loss (including via charm) results in depop.